### DISCUSSION TITLE
Can micropython be used in Trusted Execution Environments (TEEs)?

### DISCUSSION THREAD

**Discussion by bmiller59** [micropython/micropython#15901] [General]
I have a need to build a Trusted Application for Trusted Execution Environments (TEEs), for example with [OP-TEE](https://optee.readthedocs.io/en/latest/index.html). What would be involved in getting micropython, or perhaps a workable subset of it, to run in this environment?

**Comment by vshymanskyy** [comment]
I'm pretty sure it will work (or you'll be able to make it work). However I don't think there any ready to use examples

**Comment by jonnor** [comment]
Probably best to try the embed port, and the embedding examples. MicroPython is a rather standard C library/app from that perspective.

**Comment by bidrohini** [comment]
For more info, you can see the OP-TEE documentation. 
https://optee.readthedocs.io/en/latest/
